solutions/cses-1641.mdx

```diff
@@ -51,3 +57,79 @@ int main(){
 	cout << "IMPOSSIBLE" << endl;
 }
 ```
+	
+</CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class SumOfThreeValues {
+	static class Group implements Comparable<Group> {
+		int value, index;
+		
+		Group (int value, int index) {
+			this.value = value;
+			this.index = index;
+		}
+		
+		@Override
+		public int compareTo (Group other) {
+			return Integer.compare(this.value, other.value);
+		}
+	}
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int x = io.nextInt();
+		Group[] data = new Group[n];
+
+		for (int i = 1; i <= n; i++) {
+			// store the input in a class that has a value and an index
+			int value = io.nextInt();
+			Group group = new Group(value, i);
+
+			data[i - 1] = group;
+		}
+
+		Arrays.sort(data);  // make the values in increasing order
+		for (int i = 0; i < n; i++) {
+			int left = 0;
+			int right = n - 1;
+
+			while (left != right) {
+				int target = x - data[i].value;
+				// if the left and the right are not i, their values sum up to the target
```

Comment:
isn't this just a repeat of the code? i think instead u could say something like if all three values sum up to the target and aren't from the same index, we have found our answer.